### PR TITLE
Added venue sorting by orderPriority

### PR DESCRIPTION
**Pull request by alexey-petrov**
Closes https://github.com/sparkletown/internal-sparkle-issues/issues/997
* added venue sorting by orderPriority including start time / duration
* covers both positive and negative orderPriority values

### CODE REVIEW

File: src/utils/event.ts [hunk 1]
```diff
@@ -88,3 +88,20 @@ export const eventTimeComparator = (a: VenueEvent, b: VenueEvent) => {
...
+
+  if (aOrderPriority === bOrderPriority) {
+    return aStartUtcSeconds < bStartUtcSeconds
+      ? -1
+      : aStartUtcSeconds > bStartUtcSeconds
+      ? 1
+      : 0;
+  } else {
+    return aOrderPriority > bOrderPriority ? -1 : 1;
+  }
```
Comment (sunny-viktoryia) [inline src/utils/event.ts:106]: `var` -> `const`
`a < b ? -1 : 1` -> `a - b`
reuse `eventTimeComparator`

```suggestion
  const aOrderPriority = a.orderPriority ?? 0;
  const bOrderPriority = b.orderPriority ?? 0;

  if (aOrderPriority === bOrderPriority) {
    return eventTimeComparator(a, b);
  } else {
    return aOrderPriority - bOrderPriority;
  }
```

**Review by sunny-viktoryia — COMMENTED** [review]
question + suggestion

File: src/utils/event.ts (hunk 1 above)
Comment (alexey-petrov) [reply src/utils/event.ts]: fixed 6df0815

File: src/components/organisms/NavBarSchedule/NavBarSchedule.tsx [hunk 2]
```diff
@@ -136,20 +132,37 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
...
         })
       );
+    const daysEvents = [
+      ...eventsWithPositivePriority,
+      ...eventsWithNoPriority.sort(eventTimeComparator).map(
+        prepareForSchedule({
+          usersEvents: userEventIds,
+        })
+      ),
+      ...eventsWithNegativePriority,
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/organisms/NavBarSchedule/NavBarSchedule.tsx]: yep, thanks!
fixed in 6df0815

File: src/components/organisms/NavBarSchedule/NavBarSchedule.tsx [hunk 3]
```diff
@@ -136,15 +132,11 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
...
-    const daysEvents = relatedVenueEvents
-      .filter(
-        isScheduleTimeshifted
-          ? isEventWithinDate(startOfSelectedDay)
-          : isEventWithinDateAndNotFinished(startOfSelectedDay)
-      )
-      .sort(eventTimeComparator)
+    const currentEvents = relatedVenueEvents;
+
+    const eventsWithFilledWithPriority = currentEvents
```
Comment (sunny-viktoryia) [inline src/components/organisms/NavBarSchedule/NavBarSchedule.tsx:137]: one `with` in the name is extra?

**Review by sunny-viktoryia — APPROVED** [review]
looks much better! 😍 
just a small typo fix and then should be good to be merged.

File: src/components/organisms/NavBarSchedule/NavBarSchedule.tsx (hunk 3 above)
Comment (alexey-petrov) [reply src/components/organisms/NavBarSchedule/NavBarSchedule.tsx]: fixed in f5317e6

**Pull request merged**